### ISSUE TITLE
trying to get ue4 to communicate with pd using your plugin?

### ISSUE THREAD

**Issue by Moykul** [monsieurgustav/UE4-OSC#7]
Hi  monsieurgustav,

It would be very nice to get your plugin working. I get a couple of different errors such as:

LogOSC:Warning: OSC Message Error: unterminated address pattern
or 
LogOSC:Warning: OSC Message Error: element size must be multiple of four

I have linked the pd file. This type of sample set up has worked in other applications. 

the thing is I can see there must be a data stream but it is not able to break it down.

http://1drv.ms/1OzWTGs

again any ideas or help would be great appreciated.

Kind regards,
Michael.

**Comment by monsieurgustav** [comment]
Hi,
I tested the plugin using the mrpeach "packOSC" example, shipped with pd-extended.
It works just fine as far as I can say...

Your patch is a bit different than the example. (using netsend vs. udpsend for example)
There may be a bug with it...
Or a platform issue? I use Windows here.

I strongly recommand the use of OSC Data Monitor:
http://www.kasperkamperman.com/blog/osc-datamonitor/

If it cannot print your OSC messages, then they are probably no OSC message !  :)

**Comment by Moykul** [comment]
Hi monsieurgustav,

Thank you for getting back to me it is very much appreciated. I too am using windows.
I had already downloaded and used the osc monitor, very useful!

I am getting osc streams into ue4 using your plugin so again thanks. Just trying to find a way of breaking out the data streams.

Best regards,

Michael

**Comment by monsieurgustav** [comment]
Did you see the example files provided with the plugin ?
They could inspire you.

To read data from an OSC message, you basically have 2 options:
- chaining "Pop*" nodes, the 1st pop in the chain giving the 1st OSC parameter. (easy)
- using the standard "Get" node to get the ith parameter from the "Data" array, and a "As*" node to convert it to the correct type. (efficient)

Do you still have some "OSC Message Error" in the log?

**Comment by Moykul** [comment]
Hi monsieurgustav,

I am not getting the error message, I moved off pd having spent some time trying to figure out the element data problem. I will look at it again when I have time as it can be very useful for very quick prototyping. As I mentioned I was seeing the that data was coming through.

I have now been able to get a cube moving with you plugin which is great!!!
and now I need to dig deeper into the osc data stream. Once I have this working I will build a little demo and tutorial. 

Again, thank you for your help,

Michael
